### PR TITLE
Make DirExists("") throw an error.

### PR DESCRIPTION
**Pull request by psychonic**
Apparently some (very few) plugins were relying on some undocumented behavior regarding passing an empty string to DirExists. SM was returning true, assuming that it referred to the current directory.

This change causes an error to be thrown in the case of an empty string being passed, similar to what we do for OpenDirectory. "." is still valid to explicitly work with the current directory.

### CODE REVIEW

File: core/logic/smn_filesystem.cpp [hunk 1]
```diff
@@ -616,7 +616,12 @@ static cell_t sm_DirExists(IPluginContext *pContext, const cell_t *params)
 {
 	char *name;
 	pContext->LocalToString(params[1], &name);
-	
+
+	if (!name[0])
+	{
+		return pContext->ThrowNativeError("Invalid path. An empty path string is not valid, use "." to refer to the current working directory.");
```
Comment (KyleSanderson) [inline core/logic/smn_filesystem.cpp]: \" or '.

**Comment by KyleSanderson** [comment]
I still don't think it's ideal we throw an error on DirExists anywhere, but this is better then silent failures.

**Comment by psychonic** [comment]
Invalid input seems like a great reason for throwing an error. After all, you cannot name a directory "" on any common file system that I'm aware of.

**Comment by dvander** [comment]
:ship:
1. :egg:
2. :egg:
3. :egg:

**Pull request merged**